src/ir/item.rs

```diff
@@ -1443,10 +1444,28 @@ impl ItemCanonicalPath for Item {
     fn namespace_aware_canonical_path(&self,
                                       ctx: &BindgenContext)
                                       -> Vec<String> {
-        let path = self.canonical_path(ctx);
+        let mut path = self.canonical_path(ctx);
+        let mut is_constified_module_enum = false;
+        if let ItemKind::Type(ref type_) = self.kind {
+            if let Some(ref type_) = type_.safe_canonical_type(ctx) {
+                if let TypeKind::Enum(ref enum_) = *type_.kind() {
+                    if enum_.is_constified_enum_module(ctx, self) {
+                        // Type alias is inside a module
+                        is_constified_module_enum = true;
+                    }
+                }
+            }
+        }
         if ctx.options().enable_cxx_namespaces {
+            if is_constified_module_enum {
+                path.push(CONSTIFIED_ENUM_MODULE_REPR_NAME.into());
```

Comment:
I think the check should be in `namespace_aware_canonical_path`. If `canonical_path` appends `::Type` to the path, then `namespace_aware_canonical_path` will have to also check if the item is a constified module enum. Both functions would need to be changed instead of just `namespace_aware_canonical_path`.